### ISSUE TITLE
[BUG] Leaving SteamVR causes X to crash.

### ISSUE THREAD

**Issue by CarloWood** [ValveSoftware/SteamVR-for-Linux#310]
Xorg server crashes when I click 'Quit' or close the SteamVR window displaying "Please plug in your headset".

**To Reproduce**
1. Do not have a headset plugged in (not sure yet if that is relevant: I never had one plugged in).
2. Launch SteamVR
3. Close the popped up window with "Please plug in your headset"
4. The desktop vanishes, screens go black and then X restarts.

**Expected behavior**
A clean exit with all the rest of my desktop intact.

**Observed Behavior**
X crashed and leaves the following in the logs:

[137649.551] (EE)
[137649.551] (EE) Backtrace:
[137649.561] (EE) 0: /usr/lib/xorg/Xorg (xorg_backtrace+0x4d) [0x555e118feacd]
[137649.561] (EE) 1: /usr/lib/xorg/Xorg (0x555e11746000+0x1bc869) [0x555e11902869]
[137649.561] (EE) 2: /lib/x86_64-linux-gnu/libpthread.so.0 (0x7ff3f8720000+0x12890) [0x7ff3f8732890]
[137649.561] (EE) 3: /usr/lib/x86_64-linux-gnu/nvidia/xorg/libglxserver_nvidia.so (0x7ff3f3d98000+0x8b3c8d) [0x7ff3f464bc8d]
[137649.561] (EE)
[137649.561] (EE) Segmentation fault at address 0x10
[137649.561] (EE)
Fatal server error:
[137649.561] (EE) Caught signal 11 (Segmentation fault). Server aborting

**System Information:**
- Distribution: Ubuntu 18.04
- SteamVR version: 1584145096 (../Steam/steamapps/common/SteamVR/bin/version.txt)
- Steam client version: Help ->  About Steam -> Built: Feb 11 2020, at 19:27:00 
- Opted into Steam client beta?: Yes (non-beta does the same)
- Graphics driver version: 440.64
- Gist for SteamVR System Information: [See [instructions](https://github.com/ValveSoftware/SteamVR-for-Linux/blob/master/CapturingSystemReport.md)]
 I can't follow those instructions since SteamVR does have a status window yet.
However, in steam I get the following:

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD FX(tm)-8150 Eight-Core Processor           
    CPU Family:  0x15
    CPU Model:  0x1
    CPU Stepping:  0x2
    CPU Type:  0x0
    Speed:  3905 Mhz
    8 logical processors
    8 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Ubuntu 18.04.4 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.15.0-88-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11906000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime_0.20200204.0

Video Card:
    Driver:  NVIDIA Corporation GeForce RTX 2080 SUPER/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 440.64
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x10de
    DeviceID:  0x1e81
    Revision Not Detected
    Number of Monitors:  3
    Number of Logical Video Cards:  2
    Primary Display Resolution:  1920 x 1200
    Desktop Resolution: 5520 x 1200
    Primary Display Size: 20.39" x 12.76" (24.02" diag)
                                            51.8cm x 32.4cm (61.0cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 8192 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC892

Memory:
    RAM:  32176 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  3312506 Mb
    Largest Free Hard Disk Block:  390755 Mb

Steam Linux Runtime Information:
{
  "can-write-uinput" : true,
  "steam-installation" : {
    "path" : "/opt/verylarge/Steam",
    "data_path" : "/opt/verylarge/Steam",
    "issues" : [
    ]
  },
  "runtime" : {
    "path" : "/opt/verylarge/Steam/ubuntu12_32/steam-runtime",
    "version" : "0.20200204.0",
    "issues" : [
    ],
    "pinned_libs_32" : {
      "list" : [
        " 23076867      4 drwxrwxr-x   2 carlo    carlo        4096 feb 26 02:00 pinned_libs_32",
        " 23075971      4 -rw-rw-r--   1 carlo    carlo          78 feb 26 02:00 pinned_libs_32/system_libopus.so.0",
        " 23075966      4 -rw-rw-r--   1 carlo    carlo          75 feb 26 02:00 pinned_libs_32/system_libdbus-1.so.3",
        " 23075968      4 lrwxrwxrwx   1 carlo    carlo          97 feb 26 02:00 pinned_libs_32/libgtk-x11-2.0.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgtk-x11-2.0.so.0.2400.10",
        " 23075965      4 lrwxrwxrwx   1 carlo    carlo          84 feb 26 02:00 pinned_libs_32/libdbus-1.so.3 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libdbus-1.so.3.5.8",
        " 23075969      4 -rw-rw-r--   1 carlo    carlo          96 feb 26 02:00 pinned_libs_32/system_libgtk-x11-2.0.so.0",
        " 23075970      4 lrwxrwxrwx   1 carlo    carlo          86 feb 26 02:00 pinned_libs_32/libopus.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libopus.so.0.7.0",
        " 23075967      0 -rw-rw-r--   1 carlo    carlo           0 feb 26 02:00 pinned_libs_32/has_pins",
        " 23075973      4 -rw-rw-r--   1 carlo    carlo          90 feb 26 02:00 pinned_libs_32/system_libpangox-1.0.so.0",
        " 23075972      4 lrwxrwxrwx   1 carlo    carlo          95 feb 26 02:00 pinned_libs_32/libpangox-1.0.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpangox-1.0.so.0.3000.0"
      ]
    },
    "pinned_libs_64" : {
      "list" : [
        " 23076868      4 drwxrwxr-x   2 carlo    carlo        4096 feb 26 02:00 pinned_libs_64",
        " 23075974      4 lrwxrwxrwx   1 carlo    carlo          91 feb 26 02:00 pinned_libs_64/libGLU.so.1 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libGLU.so.1.3.08004",
        " 23075977      4 lrwxrwxrwx   1 carlo    carlo          92 feb 26 02:00 pinned_libs_64/libopusfile.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libopusfile.so.0.4.4",
        " 23075990      4 -rw-rw-r--   1 carlo    carlo         101 feb 26 02:00 pinned_libs_64/system_libdbusmenu-glib.so.4",
        " 23075980      4 -rw-rw-r--   1 carlo    carlo          82 feb 26 02:00 pinned_libs_64/system_libopus.so.0",
        " 23075978      4 -rw-rw-r--   1 carlo    carlo          56 feb 26 02:00 pinned_libs_64/system_libopusfile.so.0",
        " 23075979      4 lrwxrwxrwx   1 carlo    carlo          88 feb 26 02:00 pinned_libs_64/libopus.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libopus.so.0.7.0",
        " 23075993      0 lrwxrwxrwx   1 carlo    carlo          12 feb 26 02:00 pinned_libs_64/libcurl.so.3 -> libcurl.so.4",
        " 23075976      0 -rw-rw-r--   1 carlo    carlo           0 feb 26 02:00 pinned_libs_64/has_pins",
        " 23075981      4 lrwxrwxrwx   1 carlo    carlo          91 feb 26 02:00 pinned_libs_64/libopusurl.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libopusurl.so.0.4.4",
        " 23075982      4 -rw-rw-r--   1 carlo    carlo          54 feb 26 02:00 pinned_libs_64/system_libopusurl.so.0",
        " 23075983      4 lrwxrwxrwx   1 carlo    carlo          97 feb 26 02:00 pinned_libs_64/libdbusmenu-gtk.so.4 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-gtk.so.4.0.13",
        " 23075984      4 -rw-rw-r--   1 carlo    carlo          99 feb 26 02:00 pinned_libs_64/system_libdbusmenu-gtk.so.4",
        " 23075985      4 lrwxrwxrwx   1 carlo    carlo          88 feb 26 02:00 pinned_libs_64/libcurl.so.4 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libcurl.so.4.2.0",
        " 23075975      4 -rw-rw-r--   1 carlo    carlo          80 feb 26 02:00 pinned_libs_64/system_libGLU.so.1",
        " 23075986      4 -rw-rw-r--   1 carlo    carlo          82 feb 26 02:00 pinned_libs_64/system_libcurl.so.4",
        " 23075992      4 -rw-rw-r--   1 carlo    carlo          94 feb 26 02:00 pinned_libs_64/system_libpangox-1.0.so.0",
        " 23075988      4 -rw-rw-r--   1 carlo    carlo          90 feb 26 02:00 pinned_libs_64/system_libSDL2-2.0.so.0",
        " 23075987      4 lrwxrwxrwx   1 carlo    carlo          93 feb 26 02:00 pinned_libs_64/libSDL2-2.0.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0.10.0",
        " 23075989      4 lrwxrwxrwx   1 carlo    carlo          98 feb 26 02:00 pinned_libs_64/libdbusmenu-glib.so.4 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-glib.so.4.0.13",
        " 23075991      4 lrwxrwxrwx   1 carlo    carlo          97 feb 26 02:00 pinned_libs_64/libpangox-1.0.so.0 -> /opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libpangox-1.0.so.0.3000.0"
      ]
    }
  },
  "os-release" : {
    "id" : "ubuntu",
    "id_like" : [
      "debian"
    ],
    "name" : "Ubuntu",
    "pretty_name" : "Ubuntu 18.04.4 LTS",
    "version_id" : "18.04",
    "version_codename" : "bionic"
  },
  "architectures" : {
    "i386-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /usr/lib/x86_64-linux-gnu/libvulkan_intel.so: wrong ELF class: ELFCLASS64\nWARNING: [Loader Message] Code 0 : loader_icd_scan: Can not find 'ICD' object in ICD JSON file /usr/share//vulkan/icd.d/nvidia_layers.json.  Skipping ICD JSON\nERROR: [Loader Message] Code 0 : /usr/lib/x86_64-linux-gnu/libvulkan_radeon.so: wrong ELF class: ELFCLASS64\nERROR: [Loader Message] Code 0 : /opt/verylarge/Steam/ubuntu12_64/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS64\n",
          "renderer" : "GeForce RTX 2080 SUPER",
          "version" : "1.1.119 (device 10de:1e81) (driver 440.256.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce RTX 2080 SUPER/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce RTX 2080 SUPER/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "messages" : "Wflinfo error: failed to get function pointer for glGetError\n",
          "renderer" : null,
          "version" : null,
          "library-vendor" : "glvnd",
          "issues" : [
            "cannot-load"
          ],
          "exit-status" : 1
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/opt/verylarge/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/dri/dummy_drv_video.so"
        }
      ]
    },
    "x86_64-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "WARNING: [Loader Message] Code 0 : loader_icd_scan: Can not find 'ICD' object in ICD JSON file /usr/share/vulkan/icd.d/nvidia_layers.json.  Skipping ICD JSON\nERROR: [Loader Message] Code 0 : /opt/verylarge/Steam/ubuntu12_32/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS32\n",
          "renderer" : "GeForce RTX 2080 SUPER",
          "version" : "1.1.119 (device 10de:1e81) (driver 440.256.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce RTX 2080 SUPER/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce RTX 2080 SUPER/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce RTX 2080 SUPER/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/dummy_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r600_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/s3g_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/vdpau_drv_video.so"
        }
      ]
    }
  },
  "locale-issues" : [
  ],
  "locales" : {
    "<default>" : {
      "resulting-name" : "LC_CTYPE=en_US.UTF-8;LC_NUMERIC=nl_NL.UTF-8;LC_TIME=nl_NL.UTF-8;LC_COLLATE=en_US.UTF-8;LC_MONETARY=nl_NL.UTF-8;LC_MESSAGES=en_US.UTF-8;LC_PAPER=nl_NL.UTF-8;LC_NAME=nl_NL.UTF-8;LC_ADDRESS=nl_NL.UTF-8;LC_TELEPHONE=nl_NL.UTF-8;LC_MEASUREMENT=nl_NL.UTF-8;LC_IDENTIFICATION=nl_NL.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "C" : {
      "resulting-name" : "C",
      "charset" : "ANSI_X3.4-1968",
      "is_utf8" : false
    },
    "C.UTF-8" : {
      "resulting-name" : "C.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "en_US.UTF-8" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    }
  },
  "egl" : {
    "icds" : [
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/10_nvidia.json",
        "library_path" : "libEGL_nvidia.so.0"
      },
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
        "library_path" : "libEGL_mesa.so.0"
      }
    ]
  },
  "vulkan" : {
    "icds" : [
      {
        "json_path" : "/usr/share/vulkan/icd.d/nvidia_icd.json",
        "library_path" : "libGLX_nvidia.so.0",
        "api_version" : "1.1.119"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/intel_icd.x86_64.json",
        "library_path" : "/usr/lib/x86_64-linux-gnu/libvulkan_intel.so",
        "api_version" : "1.1.102"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/nvidia_layers.json",
        "error-domain" : "g-io-error-quark",
        "error-code" : 0,
        "error" : "No \"ICD\" object in \"/usr/share/vulkan/icd.d/nvidia_layers.json\""
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/radeon_icd.x86_64.json",
        "library_path" : "/usr/lib/x86_64-linux-gnu/libvulkan_radeon.so",
        "api_version" : "1.1.107"
      }
    ]
  }
}

The Steam Linux Runtime container tool is not installed.
-------------

**Screenshots**

**Additional context**

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" section in their replies.

**Comment by kedodrill** [comment]
What exactly do you mean by "Do not have a headset plugged in"? The SteamVR button does not show up for me if the USB connector is not plugged in (I am using a Valve Index so it could be different for a Vive), so I can't launch it like you're supposed to from Steam unless it's plugged in via USB.

That button for SteamVR shouldn't even show up for you unless you have a headset plugged in...are you launching it some other way?

**Comment by CarloWood** [comment]
On Mon, 16 Mar 2020 08:13:58 -0700
Kenneth Dodrill <notifications@github.com> wrote:

> What exactly do you mean by "Do not have a headset plugged in"?

As in, do not own a headset.

> The
> SteamVR button does not show up for me if the USB connector is not
> plugged in (I am using a Valve Index so it could be different for a
> Vive), so I can't launch it like you're supposed to from Steam unless
> it's plugged in via USB.
> 
> That button for SteamVR shouldn't even show up for you unless you
> have a headset plugged in...are you launching it some other way?

I installed SteamVR - you can find it in the list of Tools in steam
and Launch it. No headset needs to be plugged in.  When you do, a
small window will popup asking to plug in the headset.


-- 
Carlo Wood <carlo@alinoe.com>

**Comment by kedodrill** [comment]
@CarloWood I see, that makes more sense. You may want to change the title to something like:

"Xorg crashes when quitting SteamVR without a headset plugged in"

Just a bit more specific.

**Comment by CarloWood** [comment]
Well I didn't know if that was because I didn't have an Index plugged in. Currently I have an Index and have only seen one crash so far - can't remember if the headset was plugged in at that moment. So ok, I'll update the title.

**Comment by CarloWood** [comment]
I asked a friend with linux and an Index,

Save anything that is important :).. Then do the following: power off the HMD, shut down SteamVR and steam.  Then start steam, then start SteamVR - it should say that the headset can't be found and say "plug it in and click here to reset the headset" or something like that.  Power on the HMD.. wait a second or what and then click on the 'reset headset' link in that pop up.

His X crashed too.  This is reproducable.

I think this makes this bug more urgent since it really shouldn't be crashing when you click on 'reset headset' after doing exactly what the popup said.

**Comment by reddraggone9** [comment]
I appear to be running into the same crash, though in my case I do have my Index plugged in, it's just not recognized for whatever reason (which is another issue). Clicking "reset headset" crashes my session, sending me back to GDM's login screen. After logging back in, I found this in journalctl:
```
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE)
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) Backtrace:
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) 0: /usr/lib/Xorg (xorg_backtrace+0x4d) [0x558ca770c76d]
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) 1: /usr/lib/Xorg (0x558ca7659000+0xa9cf8) [0x558ca7702cf8]
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) 2: /usr/lib/libc.so.6 (0x7f44eb873000+0x3bd70) [0x7f44eb8aed70]
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) 3: /usr/lib/nvidia/xorg/libglxserver_nvidia.so (0x7f44e8f38000+0x8f7d2d) [0x7f44e982fd2d]
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE)
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) Segmentation fault at address 0x10
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE)
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: Fatal server error:
Mar 26 09:19:12 farnsworth /usr/lib/gdm-x-session[8649]: (EE) Caught signal 11 (Segmentation fault). Server aborting
```

- Distribution: Arch
- SteamVR version: 1.10.32
- Steam client version: Feb 11 2020, at 19:27:00
- Opted into Steam client beta?: No
- Graphics driver version: NVIDIA 440.64
<details>
<summary>Steam system information</summary>

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    CPU Family:  0x6
    CPU Model:  0x5e
    CPU Stepping:  0x3
    CPU Type:  0x0
    Speed:  4200 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.5.13-arch1-1
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12007000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime_0.20200204.0

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1080/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 440.64
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x10de
    DeviceID:  0x1b80
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1440
    Desktop Resolution: 2560 x 1440
    Primary Display Size: 21.77" x 12.24" (24.96" diag)
                                            55.3cm x 31.1cm (63.4cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 8192 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Realtek ALC1150

Memory:
    RAM:  31987 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.utf-8
    Total Hard Disk Space Available:  1801103 Mb
    Largest Free Hard Disk Block:  820562 Mb

Steam Linux Runtime Information:
{
  "can-write-uinput" : true,
  "steam-installation" : {
    "path" : "/home/lee/.local/share/Steam",
    "data_path" : "/home/lee/.local/share/Steam",
    "issues" : [
    ]
  },
  "runtime" : {
    "path" : "/home/lee/.local/share/Steam/ubuntu12_32/steam-runtime",
    "version" : "0.20200204.0",
    "issues" : [
    ],
    "pinned_libs_32" : {
      "list" : [
        " 14024726      4 drwxr-xr-x   2 lee      lee          4096 Feb 15 12:34 pinned_libs_32",
        " 14024743      4 -rw-r--r--   1 lee      lee            52 Feb 15 12:34 pinned_libs_32/system_libcurl.so.4",
        " 14024745      4 -rw-r--r--   1 lee      lee            71 Feb 15 12:34 pinned_libs_32/system_libdbusmenu-glib.so.4",
        " 14024747      4 -rw-r--r--   1 lee      lee            69 Feb 15 12:34 pinned_libs_32/system_libdbusmenu-gtk.so.4",
        " 14024736      4 lrwxrwxrwx   1 lee      lee            94 Feb 15 12:34 pinned_libs_32/libjack.so.0 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libjack.so.0.1.0",
        " 14024728      4 lrwxrwxrwx   1 lee      lee            92 Feb 15 12:34 pinned_libs_32/libdbus-1.so.3 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libdbus-1.so.3.5.8",
        " 14024738      4 lrwxrwxrwx   1 lee      lee            97 Feb 15 12:34 pinned_libs_32/libGLU.so.1 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libGLU.so.1.3.08004",
        " 14024737      4 -rw-r--r--   1 lee      lee            53 Feb 15 12:34 pinned_libs_32/system_libjack.so.0",
        " 14024740      4 lrwxrwxrwx   1 lee      lee           105 Feb 15 12:34 pinned_libs_32/libgtk-x11-2.0.so.0 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgtk-x11-2.0.so.0.2400.10",
        " 14024742      4 lrwxrwxrwx   1 lee      lee            94 Feb 15 12:34 pinned_libs_32/libcurl.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libcurl.so.4.2.0",
        " 14024729      4 -rw-r--r--   1 lee      lee            58 Feb 15 12:34 pinned_libs_32/system_libdbus-1.so.3",
        " 14024739      4 -rw-r--r--   1 lee      lee            50 Feb 15 12:34 pinned_libs_32/system_libGLU.so.1",
        " 14024758      0 lrwxrwxrwx   1 lee      lee            12 Feb 15 12:34 pinned_libs_32/libcurl.so.3 -> libcurl.so.4",
        " 14024744      4 lrwxrwxrwx   1 lee      lee           104 Feb 15 12:34 pinned_libs_32/libdbusmenu-glib.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libdbusmenu-glib.so.4.0.13",
        " 14024731      4 lrwxrwxrwx   1 lee      lee            91 Feb 15 12:34 pinned_libs_32/libudev.so.0 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libudev.so.0.13.0",
        " 14024730      0 -rw-r--r--   1 lee      lee             0 Feb 15 12:34 pinned_libs_32/has_pins",
        " 14024746      4 lrwxrwxrwx   1 lee      lee           103 Feb 15 12:34 pinned_libs_32/libdbusmenu-gtk.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libdbusmenu-gtk.so.4.0.13",
        " 14024741      4 -rw-r--r--   1 lee      lee            70 Feb 15 12:34 pinned_libs_32/system_libgtk-x11-2.0.so.0",
        " 14024732      4 -rw-r--r--   1 lee      lee            55 Feb 15 12:34 pinned_libs_32/system_libudev.so.0"
      ]
    },
    "pinned_libs_64" : {
      "list" : [
        " 14024727      4 drwxr-xr-x   2 lee      lee          4096 Feb 15 12:34 pinned_libs_64",
        " 14024753      4 -rw-r--r--   1 lee      lee            48 Feb 15 12:34 pinned_libs_64/system_libcurl.so.4",
        " 14024755      4 -rw-r--r--   1 lee      lee            67 Feb 15 12:34 pinned_libs_64/system_libdbusmenu-glib.so.4",
        " 14024757      4 -rw-r--r--   1 lee      lee            65 Feb 15 12:34 pinned_libs_64/system_libdbusmenu-gtk.so.4",
        " 14024748      4 lrwxrwxrwx   1 lee      lee            96 Feb 15 12:34 pinned_libs_64/libjack.so.0 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libjack.so.0.1.0",
        " 14024750      4 lrwxrwxrwx   1 lee      lee            99 Feb 15 12:34 pinned_libs_64/libGLU.so.1 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libGLU.so.1.3.08004",
        " 14024749      4 -rw-r--r--   1 lee      lee            49 Feb 15 12:34 pinned_libs_64/system_libjack.so.0",
        " 14024752      4 lrwxrwxrwx   1 lee      lee            96 Feb 15 12:34 pinned_libs_64/libcurl.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libcurl.so.4.2.0",
        " 14024751      4 -rw-r--r--   1 lee      lee            46 Feb 15 12:34 pinned_libs_64/system_libGLU.so.1",
        " 14024759      0 lrwxrwxrwx   1 lee      lee            12 Feb 15 12:34 pinned_libs_64/libcurl.so.3 -> libcurl.so.4",
        " 14024754      4 lrwxrwxrwx   1 lee      lee           106 Feb 15 12:34 pinned_libs_64/libdbusmenu-glib.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-glib.so.4.0.13",
        " 14024733      4 lrwxrwxrwx   1 lee      lee            93 Feb 15 12:34 pinned_libs_64/libudev.so.0 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu/libudev.so.0.13.0",
        " 14024735      0 -rw-r--r--   1 lee      lee             0 Feb 15 12:34 pinned_libs_64/has_pins",
        " 14024756      4 lrwxrwxrwx   1 lee      lee           105 Feb 15 12:34 pinned_libs_64/libdbusmenu-gtk.so.4 -> /home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-gtk.so.4.0.13",
        " 14024734      4 -rw-r--r--   1 lee      lee            51 Feb 15 12:34 pinned_libs_64/system_libudev.so.0"
      ]
    }
  },
  "os-release" : {
    "id" : "arch",
    "name" : "Arch Linux",
    "pretty_name" : "Arch Linux",
    "build_id" : "rolling"
  },
  "architectures" : {
    "i386-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /home/lee/.local/share/Steam/ubuntu12_64/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS64\n",
          "renderer" : "GeForce GTX 1080",
          "version" : "1.1.119 (device 10de:1b80) (driver 440.256.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib32/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib32/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/dri/dummy_drv_video.so"
        }
      ]
    },
    "x86_64-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /home/lee/.local/share/Steam/ubuntu12_32/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS32\n",
          "renderer" : "GeForce GTX 1080",
          "version" : "1.1.119 (device 10de:1b80) (driver 440.256.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce GTX 1080/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.64",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/dri/vmwgfx_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/usr/lib/dri/iHD_drv_video.so"
        },
        {
          "library_path" : "/home/lee/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/dri/dummy_drv_video.so"
        }
      ]
    }
  },
  "locale-issues" : [
    "c-utf8-missing",
    "i18n-supported-missing"
  ],
  "locales" : {
    "<default>" : {
      "resulting-name" : "en_US.utf-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "C" : {
      "resulting-name" : "C",
      "charset" : "ANSI_X3.4-1968",
      "is_utf8" : false
    },
    "C.UTF-8" : {
      "error-domain" : "srt-locale-error-quark",
      "error-code" : 0,
      "error" : "No such file or directory"
    },
    "en_US.UTF-8" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    }
  },
  "egl" : {
    "icds" : [
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/10_nvidia.json",
        "library_path" : "libEGL_nvidia.so.0"
      },
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
        "library_path" : "libEGL_mesa.so.0"
      }
    ]
  },
  "vulkan" : {
    "icds" : [
      {
        "json_path" : "/usr/share/vulkan/icd.d/nvidia_icd.json",
        "library_path" : "libGLX_nvidia.so.0",
        "api_version" : "1.1.119"
      }
    ]
  }
}

The Steam Linux Runtime container tool is not installed.
```
</details>

**Comment by betam4x** [comment]
I wanted to add a bit to the above discussion.

If no headset is plugged in and you launch SteamVR and then close it, X will crash on NVIDIA systems with the latest drivers. My config is similar to @reddraggone9 except I'm running KDE instead of GNOME.
